pkg/stub/handler.go

```diff
@@ -2,96 +2,401 @@ package stub
+
+	restconfig  *restclient.Config
+	tempclient  *templatev1client.TemplateV1Client
+	imageclient *imagev1client.ImageV1Client
+	coreclient  *corev1client.CoreV1Client
+
+	imageclientwrapper    ImageStreamClientWrapper
+	templateclientwrapper TemplateClientWrapper
+	secreteclientwrapper  SecretClientWrapper
+
+	fileimagegetter    ImageStreamFromFileGetter
+	filetemplategetter TemplateFromFileGetter
+	filefinder         ResourceFileLister
+
+	namespace string
+
+	skippedTemplates    map[string]bool
+	skippedImagestreams map[string]bool
+}
+
+func (h *Handler) manageDockerCfgSecret(deleted bool) error {
+	err := h.secreteclientwrapper.Delete("openshift", h.registrySecret.Name, &metav1.DeleteOptions{})
+	if err != nil && !kerrors.IsNotFound(err) {
+		return h.processError(v1alpha1.SecretUpdateFailed, err, "failed to delete before create dockerconfig secret the openshift namespace: %v")
+	}
+	if deleted {
+		logrus.Printf("registry dockerconfig secret %s was deleted", h.registrySecret.Name)
+		h.registrySecret = nil
+		h.operatorCfg.Status.SecretCreated = false
+		h.sdkwrapper.Update()
+	} else {
+		secretToCreate := corev1.Secret{}
+		h.registrySecret.DeepCopyInto(&secretToCreate)
+		secretToCreate.Namespace = ""
+		secretToCreate.ResourceVersion = ""
+
+		_, err = h.secreteclientwrapper.Create("openshift", &secretToCreate)
+		if err != nil {
+			return h.processError(v1alpha1.SecretUpdateFailed, err, "failed to create registry dockerconfig secret in openshift namespace : %v")
+		}
+		logrus.Printf("created/updated dockerconfig secret %s in openshift namespace", h.operatorCfg.Spec.RegistryCredentials.Name)
+
+		h.operatorCfg.Status.SecretCreated = true
+		h.sdkwrapper.Update()
+		// TODO - do we want to duplicate "oc secrets link default <pull_secret_name> --for=pull"
+		// via client-go logic for the openshift namespace's default SA here as well ??
+
+	}
+
+	status := v1alpha1.ResourceCondition{
+		Type: v1alpha1.SecretUpdateComplete,
+		Time: kapis.Now(),
+	}
+	h.operatorCfg.Status.Conditions = append(h.operatorCfg.Status.Conditions, status)
+	err = h.sdkwrapper.Update()
+	if err != nil {
+		if !kerrors.IsConflict(err) {
+			return h.processError(v1alpha1.SamplesUpdateFailed, err, "failed adding success condition to config: %v")
+		}
+		logrus.Println("got conflict error on success status update, just going to forgo this time vs. retry")
+	}
+
+	return nil
 }
 
 func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
-	switch o := event.Object.(type) {
+	switch event.Object.(type) {
+	case *corev1.Secret:
+		dockercfgSecret, _ := event.Object.(*corev1.Secret)
+
+		if h.operatorCfg == nil {
+			h.registrySecret = dockercfgSecret
+			return nil
+		}
+
+		if h.operatorCfg.Spec.RegistryCredentials.Name == dockercfgSecret.Name {
+			if h.registrySecret != nil &&
+				h.registrySecret.ResourceVersion == dockercfgSecret.ResourceVersion &&
+				h.operatorCfg.Status.SecretCreated &&
+				!event.Deleted {
+				return nil
```

Comment:
since this could be a relist event and you might have missed the "modified" event when the secret was updated, I think you have to update the secret, just in case.

Reply:
wouldn't the resource version change in that case ?  and if so, the resource version comparison would catch that ...

Reply:
yeah, sorry, I guess it would. but see my new comment about not holding registrySecret as a pointer.